src/discof/repair/fd_repair_tile.c

```diff
@@ -498,6 +801,40 @@ after_frag( fd_repair_tile_ctx_t * ctx,
     return;
   }
 
+  if( FD_UNLIKELY( in_kind==IN_KIND_SIGN ) ) {
+    /* Nonce was packed into sig, so we need to unpack it */
+    ulong response_nonce = sig >> 32;
+
+    /* Look up the pending request by nonce. Since the repair_sign links are
+       reliable, the incoming sign_repair fragments represent a complete
+       set of the previously sent outgoing messages. However, with
+       multiple sign tiles, the responses may not arrive in order. */
+
+    /* Find which sign tile sent this response and increment its credits */
+    for( uint i = 0; i < ctx->repair_sign_cnt; i++ ) {
+      if( ctx->repair_sign_out_ctx[i].in_idx == in_idx ) {
+        if( ctx->repair_sign_out_ctx[i].credits < ctx->repair_sign_out_ctx[i].max_credits ) {
+          ctx->repair_sign_out_ctx[i].credits++;
+        }
+        break;
+      }
+    }
+
+    fd_repair_pending_sign_req_t * pending = fd_repair_pending_sign_req_map_ele_query( ctx->pending_sign_req_map, &response_nonce, NULL, ctx->pending_sign_req_pool );
+    if( FD_LIKELY( pending ) ) {
+      fd_memcpy( pending->buf + pending->sig_offset, ctx->buffer, 64UL );
+      ulong tsorig = fd_frag_meta_ts_comp( fd_tickcount() );
+      uint src_ip4_addr = 0U;
+      send_packet( ctx, stem, 1, pending->dst_ip_addr, pending->dst_port, src_ip4_addr, pending->buf, pending->buflen, tsorig );
+
+      fd_repair_pending_sign_req_map_ele_remove( ctx->pending_sign_req_map, &response_nonce, NULL, ctx->pending_sign_req_pool );
+      fd_repair_pending_sign_req_pool_ele_release( ctx->pending_sign_req_pool, pending );
+      return;
+    } else {
+      FD_LOG_WARNING(( "No pending request found for nonce %lu", response_nonce ));
+    }
+  }
```

Comment:
can you extract this out into a unit test and make sure there out-of-order receive actually works?